protocol/common/message_linkpreview.go

```diff
@@ -288,10 +289,17 @@ func (m *Message) ConvertLinkPreviewsToProto() ([]*protobuf.UnfurledLink, error)
 			return nil, fmt.Errorf("invalid link preview, url='%s': %w", preview.URL, err)
 		}
 
-		var payload []byte
+		var thumbnailPayload []byte
+		var faviconPayload []byte
 		var err error
 		if preview.Thumbnail.DataURI != "" {
-			payload, err = images.GetPayloadFromURI(preview.Thumbnail.DataURI)
+			thumbnailPayload, err = images.GetPayloadFromURI(preview.Thumbnail.DataURI)
+			if err != nil {
+				return nil, fmt.Errorf("could not get data URI payload, url='%s': %w", preview.URL, err)
```

Comment:
We could be more specific in the error message since now the data URI could come from fetching a favicon or a thumbnail. E.g. `"could not get thumbnail data URI payload"`

Reply:
Done